lib/IRGen/GenDistributed.cpp

```diff
@@ -836,15 +838,35 @@ ArgumentDecoderInfo DistributedAccessor::findArgumentDecoder(
                        instance);
 
     decoder = instance.claimNext();
-  }
 
-  auto *decodeSIL = IGM.getSILModule().lookUpFunction(SILDeclRef(decodeFn));
-  auto *fnPtr = IGM.getAddrOfSILFunction(decodeSIL, NotForDefinition,
-                                         /*isDynamicallyReplaceable=*/false);
+    /// When using library evolution functions have another "dispatch thunk"
+    /// so we must use this instead of the decodeFn directly.
+   if (classDecl->hasResilientMetadata() && !classDecl->isFinal()) {
```

Comment:
the isFinal here is guessing tbh... I've spent a long time finding the right condition but can't seem to find it -- this _seems to_ work in practical testing in a resilient module and normal projects also are happy (cluster, tests),